### PR TITLE
Use ALPHA_EQUIV as a reconstruction tactic prior to RARE

### PR DESCRIPTION
**Pull request by ajreynol**
This generalizes the preprocessing step for when we are trying to prove the equivalence between 2 quantified formulas.

In particular, we previously always tried to show equivalence when the 2 quantified formulas had equivalent binders by showing their body was equal. We now also take into account alpha equivalence.

### CODE REVIEW

File: src/rewriter/rewrite_db_proof_cons.cpp [hunk 1]
```diff
@@ -173,6 +155,118 @@ bool RewriteDbProofCons::prove(
...
+    if (!res.isNull())
+    {
+      Assert(res.getKind() == Kind::EQUAL);
+      cdp->addStep(res, ProofRule::ALPHA_EQUIV, {}, aeArgs);
+      // Remains to prove that the result of applying alpha equivalence to the
+      // left hand side is equal to the right hand side. This may be a
+      // reflexive equality when ALPHA_EQUIV alone suffices. Note that when this
+      // occurs eq is not a free assumption in cdp. Proof generation will
+      // easily succeed in proving eq by REFL, but this will not be used
+      // in the final proof.
```
Comment (HanielB) [inline src/rewriter/rewrite_db_proof_cons.cpp:230]: what do you mean with "this will not be used in the final proof"?

Comment (ajreynol) [reply src/rewriter/rewrite_db_proof_cons.cpp:230]: Say we want to prove rewrite `(= t s)`, and we know via ALPHA_EQUIV `(= t r)`. Then this code says it suffices to show `(= r s)`.  However, if `s` is `r`, then we have already succeeded and don't need the proof of `(= r s)`.

Comment (HanielB) [reply src/rewriter/rewrite_db_proof_cons.cpp:230]: I see. Can you add this example to the doc? It'd help making it clearer.

Comment (ajreynol) [reply src/rewriter/rewrite_db_proof_cons.cpp:230]: Done

**Review by HanielB — APPROVED** [review]

**Pull request merged**